tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Hooks;
+
+function getimagesize($filename = null, array &$imageinfo = null) {
+	return ImageHooksTest::$dimensions ?: getimagesize($filename);
+}
+
+use OCA\MediaMetadata\Services\ImageDimension;
+use OCA\MediaMetadata\Services\ImageDimensionMapper;
+use OCA\MediaMetadata\Tests\TestCase;
+use OCP\ILogger;
+use OCP\IDb;
+use OCP\Files\File;
+use OCP\Files\Folder;
+
+/**
+ * Class ImageHooksTest
+ *
+ * @package OCA\MediaMetadata\Hooks
+ */
+class ImageHooksTest extends TestCase {
+	/**
+	 * @var \OCA\MediaMetadata\Hooks\ImageHooks
+	 */
+	protected $imagehooks;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OC\Files\Node\Root
+	 */
+	protected $root;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimensionMapper
+	 */
+	protected $mapper;
+	/**
+	 * @var ILogger
+	 */
+	protected $logger;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCP\IDb
+	 */
+	protected $db;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimension
+	 */
+	protected $imageDimension;
+
+	/**
+	 * @var array $dimensions that will be returned by getimagesize()
+	 */
+	public static $dimensions;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->root = $this->getMockBuilder('OC\Files\Node\Root')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->mapper = $this->mockImageDimensionMapper();
+		$this->logger = $this->getMockBuilder('\OCP\ILogger')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->db = $this->getMockBuilder('\OCP\IDb')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->imageDimension = $this->mockImageDimension();
+
+		$app = new \OCA\MediaMetadata\AppInfo\Application();
+		$container = $app->getContainer();
+		$this->imagehooks = new ImageHooks(
+			$this->root,
+			$this->mapper,
+			$container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory')
```

Comment:
Yeah, it was giving an error and then I tried this and it didn't give the error again.